firewood/src/merkle/proof.rs

```diff
@@ -352,31 +353,31 @@ impl<N: AsRef<[u8]> + Send> Proof<N> {
                 }
                 Some(p) => {
                     // Return when reaching the end of the key.
-                    if key_index == chunks.len() {
-                        #[allow(clippy::indexing_slicing)]
-                        (cur_key = &chunks[key_index..]);
+                    if key_nibbles.size_hint().0 == 0 {
```

Comment:
Nit: the other case is much easier to handle earlier (find the comment "The trie doesn't contain the key") so it should be moved here, reducing the indendation and complexity of this block.

Reply:
I think doing this made the code a little harder to review FYI, but the resultant code is much nicer... I probably should have done this in a follow up